pyro/infer/mcmc/nuts.py

```diff
@@ -352,7 +352,13 @@ def sample(self, params):
                 sum_accept_probs = sum_accept_probs + new_tree.sum_accept_probs
                 num_proposals = num_proposals + new_tree.num_proposals
 
-                if new_tree.turning or new_tree.diverging:  # stop doubling
+                # stop doubling
+                if new_tree.diverging:
+                    if self._t >= self._warmup_steps:
+                        self._num_diverging += 1
```

Comment:
This diagnostic information will be really useful. I think users also like to plot their posterior samples and see where the divergences lie. To support that, we will also need to annotate the samples somehow for later analysis. This is much easier to do in NumPyro by adding to the named tuple data structure, but I wonder what will be the best way to do this in Pyro. Maybe we can store it in a separate list inside the HMC/NUTS kernels, called `kernel._divergences`. What do you think?

Reply:
Sound reasonable for me! For now, I'll go with your proposal and will think how to let users access it with multi chains.